istioctl/cmd/istiodconfig.go

```diff
@@ -0,0 +1,331 @@
+	matched, err := regexp.MatchString(validationPattern, sslp)
+	if err != nil {
+		return nil, err
+	}
+	if !matched {
+		return nil, fmt.Errorf("pattern %s did not match", sslp)
+	}
+	fmt.Println("validation ok ", sslp)
+	scopeStackTraceLevel := strings.Split(sslp, ":")
+	ss := &scopeStackTraceLevelPair{
+		scope:    scopeStackTraceLevel[0],
+		logLevel: scopeStackTraceLevel[1],
+	}
+	return ss, nil
+}
+
+// TODO: create constructor to ensure the valid client for controlz is created
+type controlzClient struct {
+	baseUrl    *url.URL
+	httpClient *http.Client
+}
+
+func (c *controlzClient) GetScopes() ([]*scopeInfo, error) {
+	var scopeInfos []*scopeInfo
+	resp, err := c.httpClient.Get(c.baseUrl.String())
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("request not successful %s", resp.Status)
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&scopeInfos)
+	if err != nil {
+		return nil, fmt.Errorf("cannot deserialize response %s", err)
+	}
+	return scopeInfos, nil
+}
+
+func (c *controlzClient) PutScope(scope *scopeInfo) error {
+	var jsonScopeInfo bytes.Buffer
+	err := json.NewEncoder(&jsonScopeInfo).Encode(scope)
+	if err != nil {
+		return fmt.Errorf("cannot serialize scope %+v", *scope)
+	}
+	req, err := http.NewRequest(http.MethodPut, c.baseUrl.String()+"/"+scope.Name, &jsonScopeInfo)
+	if err != nil {
+		return err
+	}
+	defer req.Body.Close()
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("cannot update resource %s, got status %s", scope.Name, resp.Status)
+	}
+	return nil
+}
+
+func (c *controlzClient) GetScope(scope string) (*scopeInfo, error) {
+	var s scopeInfo
+	resp, err := http.Get(c.baseUrl.String() + "/" + scope)
+	if err != nil {
+		return &s, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return &s, fmt.Errorf("request not successful %s: ", resp.Status)
+	}
+
+	err = json.NewDecoder(resp.Body).Decode(&s)
+	if err != nil {
+		return &s, fmt.Errorf("cannot deserialize response: %s", err)
+	}
+	return &s, nil
+}
+
+func istiodLogCmd() *cobra.Command {
+	var opts clioptions.ControlPlaneOptions
+	var outputLevel = ""
+	var stackTraceLevel = ""
+	var validationPattern = `^\w+:(debug|error|warn|info|debug)`
+	var labelSelector = "app=istiod"
+
+	logCmd := &cobra.Command{
+		Use:   "log [--level <scope>:<level>][--stack-trace-level <scope>:<level>]|[-r|--reset]",
+		Short: "Manage istiod logging.",
+		Long:  "Retrieve or update logging levels of istiod components.",
+		Example: ` # Retrieve information about istiod logging levels.
+  istioctl istiod-config log [<pod-name>]
+
+  # Update levels of the specified loggers.
+  istioctl istiod-config log --level ads:debug,authorization:debug
+
+  # Reset levels of all the loggers to default value (warning).
```

Comment:
nit: warning is not default log level for istiod

Reply:
good catch thank you.